### ISSUE TITLE
neovim: migrate to neovim's new remote plugin design

### ISSUE THREAD

**Issue by fidgetingbits** [cursorless-dev/cursorless#2397]
I'm not sure yet what we'll have to change, but https://github.com/neovim/neovim/issues/27949 is tracking development that will rework how remote plugins work in neovim, so likely will affect how the node function registration works.

Once we change it will also mean we have to specify whatever version of neovim ends up supporting the new way (currently planned for 0.11) will become a requirement for using cursorless.nvim.

**Comment by saidelike** [comment]
Yes, the PR is https://github.com/neovim/node-client/pull/344 and as you mentioned it will change how we load plugins (https://github.com/neovim/node-client/issues/356) but will also allow loading command-server outside of cursorless.nvim (https://github.com/cursorless-dev/cursorless/issues/2351)

**Comment by saidelike** [comment]
Probably also worth mentioning this project https://github.com/roxma/nvim-yarp which allows running remote plugins without “:UpdateRemotePlugins”
